go-app-ussd_popi_user_data.js

```diff
@@ -319,16 +325,33 @@ go.app = function() {
                 ],
                 next: function(choice) {
                     return self.im.user
-                    // TODO: update on IdentityStore
                     .set_lang(choice.value)
                     .then(function() {
-                        self.im.user.set_answer("state_language", choice.value);
-                        return 'state_updated';
+                        return 'state_switch_lang';
                     });
                 },
             });
         });
 
+        self.add('state_switch_lang', function(name) {
+            var change_info = {
+                "registrant_id": self.im.user.answers.operator.id,
+                "action": "momconnect_change_language",
+                "data": {
+                    "language": self.im.user.answers.state_select_language,
+                    "old_language": self.im.user.answers.operator.details.lang_code
+                }
+            };
+
+            self.im.user.set_lang(self.im.user.answers.state_select_language);
```

Comment:
Sorry that I didn't see this before, but it seems like we're calling `set_lang` already in the previous state? Maybe we can remove all the `next` logic from the previous state, and just go directly to this state, which does all the logic to change the correct things?

Also, `set_lang` returns a promise, so by not waiting for that promise to be fulfilled before moving on, we don't have any guarantees that the user's language will be change by the next state, so sometimes they will get the next state in the new language, and sometimes in the old language.